src/include/sof/audio/component.h

```diff
@@ -779,27 +780,37 @@ int comp_set_state(struct comp_dev *dev, int cmd);
 static inline void component_set_nearest_period_frames(struct comp_dev *current,
 						       uint32_t rate)
 {
+	uint64_t period = current->period;
+	uint64_t frames;
+
 	/* Sample rate is in Hz and period in microseconds.
 	 * As we don't have floats use scale divider 1000000.
 	 * Also integer round up the result.
 	 * dma buffer size should align with 32bytes which can't be
 	 * compatible with current 45K adjustment. 48K is a suitable
 	 * adjustment.
 	 */
+
 	switch (rate) {
 	case 44100:
-		current->frames = 48000 * current->period / 1000000;
+		frames = 48000 * period / 1000000;
 		break;
 	case 88200:
-		current->frames = 96000 * current->period / 1000000;
+		frames = 96000 * period / 1000000;
 		break;
 	case 176400:
-		current->frames = 192000 * current->period / 1000000;
+		frames = 192000 * period / 1000000;
 		break;
 	default:
-		current->frames = ceil_divide(rate * current->period, 1000000);
+		frames = rate * period / 1000000;
```

Comment:
seems duplicated code, can we try to change as below:
	switch (rate) {
	case 44100:
		rate = 48000;
		break;
	case 88200:
		rate = 96000;
		break;
	case 176400:
		rate = 192000;
		break;
	default:
		break;

frames = rate * period / 1000000;
if(rate*period % 1000000)
    frames++;

Reply:
Now updated in V2, @btian1 please check.